### PR TITLE
Make OpsGenie API Key a templated string

### PR DESCRIPTION
**Pull request by twz123**
This is analogous of what is done for PagerDuty secrets.

Probably needs a doc update as well.

/cc @stuartnelson3

### CODE REVIEW

**Comment by brian-brazil** [comment]
My understanding is that this is not used for routing, has this changed?

**Comment by twz123** [comment]
OpsGenie has so-called Team Integrations, which are essentially team-specific versions of global (account-wide) Integrations. They allow teams to configure their incoming alert rules separately from each other. Such Team Integrations get their own API keys. So this is somewhat a special version of an Alert->Team routing.

File: notify/impl_test.go [hunk 1]
```diff
@@ -218,12 +220,28 @@ func readBody(t *testing.T, r *http.Request) string {
 }
 
 func TestOpsGenie(t *testing.T) {
+	templateDir, err := ioutil.TempDir("", "opsgenie-")
```
Comment (brian-brazil) [inline notify/impl_test.go]: Why are you doing all this setup? You can test this like the others.

Comment (twz123) [reply notify/impl_test.go]: Ah, you mean to create a file in testdata? Didn't occur to me :see_no_evil: Will do!

Comment (brian-brazil) [reply notify/impl_test.go]: You should't need to create any files. The use case here is that the apikey is coming from an alert label, and any random template expression is sufficient to verify that templating is being applied.

**Comment by brian-brazil** [comment]
Thanks! I'd suggest a read of https://prometheus.io/docs/operating/security/#alertmanager

**Pull request merged**